engine/resources/mount_test.go

```diff
@@ -0,0 +1,321 @@
+// Mgmt
+// Copyright (C) 2013-2018+ James Shubin and the project contributors
+// Written by James Shubin <james@shubin.ca> and the project contributors
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package resources
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+
+	fstab "github.com/deniswernert/go-fstab"
+)
+
+const fstabMock1 = `UUID=ef5726f2-615c-4350-b0ab-f106e5fc90ad / ext4 defaults 1 1` + "\n"
+
+const procMock1 = `/tmp/mount0 /mnt/proctest ext4 rw,seclabel,relatime,data=ordered 0 0` + "\n"
+
+var fstabEntryExistsTests = []struct {
+	fstabMock []byte
+	in        *fstab.Mount
+	out       bool
+}{
+	{
+		[]byte(fstabMock1),
+		&fstab.Mount{
+			Spec:    "UUID=ef5726f2-615c-4350-b0ab-f106e5fc90ad",
+			File:    "/",
+			VfsType: "ext4",
+			MntOps:  map[string]string{"defaults": ""},
+			Freq:    1,
+			PassNo:  1,
+		},
+		true,
+	},
+	{
+		[]byte(fstabMock1),
+		&fstab.Mount{
+			Spec:    "/dev/mapper/root",
+			File:    "/home",
+			VfsType: "ext4",
+			MntOps:  map[string]string{"defaults": ""},
+			Freq:    1,
+			PassNo:  1,
+		},
+		false,
+	},
+}
+
+func TestFstabEntryExists(t *testing.T) {
+	file, err := ioutil.TempFile("", "fstab")
+	if err != nil {
+		t.Errorf("error creating temp file: %s", err)
+		return
+	}
+	defer os.Remove(file.Name())
+
+	for _, test := range fstabEntryExistsTests {
+		if err := ioutil.WriteFile(file.Name(), test.fstabMock, 0644); err != nil {
+			t.Errorf("error writing fstab file: %s: %s", file.Name(), err)
+		}
+		result, err := fstabEntryExists(file.Name(), test.in)
+		if err != nil {
+			t.Errorf("error checking if fstab entry %s exists: %s", test.in.String(), err)
```

Comment:
make sure to add `return`'s when needed...

also as discussed, nit: use of `err` in Errorf should be changed to %v instead of %s throughout.